clip_text.py

```diff
@@ -6,35 +6,53 @@
 
 
 class CLIPTextEncoder(Executor):
-    """Encode text into embeddings."""
+    """Encode text into embeddings using a CLIP model.
 
-    def __init__(self,
-                 model_name: str = 'ViT-B/32',
-                 default_batch_size: int = 32,
-                 default_traversal_paths: Optional[List[str]] = None,
-                 default_device: Optional[str] = 'cpu',
-                 jit: bool = True,
-                 *args, **kwargs):
+    :param model_name: The name of one of the pre-trained CLIP models.
+        Can also be a path to a local checkpoint (a ``.pt`` file).
+    :param default_batch_size: Default batch size for encoding, used if the
+        batch size is not passed as a parameter with the request.
+    :param default_traversal_paths: Default traversal paths for encoding, used if the
+        traversal path is not passed as a parameter with the request.
+    :param default_device: The device (cpu or gpu) that the model should be on.
+    :param jit: Whether a JIT version of the model should be loaded.
+    """
+
+    def __init__(
+        self,
+        model_name: str = 'ViT-B/32',
+        default_batch_size: int = 32,
+        default_traversal_paths: Optional[List[str]] = None,
+        default_device: str = 'cpu',
+        jit: bool = True,
+        *args,
+        **kwargs
+    ):
         super().__init__(*args, **kwargs)
-        self.device = default_device or 'cpu'
+        self.device = default_device
         self.model, _ = clip.load(model_name, self.device, jit)
         self.default_traversal_paths = default_traversal_paths or ['r']
         self.default_batch_size = default_batch_size
 
     @requests
-    def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
+    def encode(
+        self, docs: Optional[DocumentArray] = None, parameters: dict = {}, **kwargs
```

Comment:
we do not need `default` values here.

Reply:
Can you explain what you mean here - should users be forced to pass in `parameters` with batch size and traversal paths on each request?

Reply:
`docs` will be `None` automatically, if not passed

`parameters` will be `{}`, if not passed

Reply:
@cristianmtr not sure about the second one - if I remove defaults and run this test
```python
def test_no_documents():
    clip_text_encoder = CLIPTextEncoder()
    test_docs = DocumentArray()
    clip_text_encoder.encode(test_docs)
    assert len(test_docs) == 0  # SUCCESS
```
it fails with

```
    def test_no_documents():
        clip_text_encoder = CLIPTextEncoder()
        test_docs = DocumentArray()
>       clip_text_encoder.encode(test_docs)

tests/unit/test_clip_text.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<jinahub.encoder.clip_text.CLIPTextEncoder object at 0x7f9448710040>, <jina.types.arrays.document.DocumentArray length=0 at 140274847252688>), kwargs = {}

    @functools.wraps(fn)
    def arg_wrapper(*args, **kwargs):
>       return fn(*args, **kwargs)
E       TypeError: encode() missing 1 required positional argument: 'parameters'
```

Reply:
Yes, but that's because you call the method yourself. I'm referring to the Runtime/Pea

Reply:
Okay thanks for clearing that up. Then I'll remove the defaults and adjust the test. 

A bit worrying that whatever PR introduced this faulty test (+ no CI) passed code review before...

Reply:
It did? Can u check if the CI is working at all? Maybe there's some misconfig there. I see that it's running now.

Reply:
There was no CI github action before - that was the problem